protocol.md

```diff
@@ -51,6 +51,151 @@ satisfied or after a timeout.
 5. **Transaction Broadcast** - The coordinator broadcasts the transaction and waits
    for confirmation in a block.
 
+Timeouts are considered a failure except during input registration, where the
+round may still proceed if a sufficient number of participants have joined.
+Failure during any phase will cause the round as a whole to fail, apart from
+output registration which unconditionally proceeds to the signing phase in
+order to assign blame. Signed inputs reveal that a user executed the protocol
+honestly, allowing faulty/malicious users to be excluded from a re-attempted
+round.
+
+Round metadata such as coordinator parameters for the WabiSabi credentials,
+information is provided in responses to `coordinator-status` requests.
+
+## Attacks
+
+Due to the nature of CoinJoin transactions users don't need to trust other
+users or the coordinator against theft, leaving denial of service and attacks
+on privacy as the main concerns.
+
+### Denial of Service
+
+A user may disrupt the protocol by failing to provide a signature for an input
+on the final transaction.
+
+Such attacks can be mitigated by banning inputs that do not honestly
+participate, which can increase the cost and liquidity requirement for a denial
+of service attack.
+
+Failure to confirm during input registration can generally be tolerated as it does not harm other users. However, failure to confirm in connection confirmation phase cannot go without punishment, otherwise an attacker could make sure only small CoinJoins to take place. And once the connection confirmation phase times out the input set is finalized, and inputs for which a signature is not the subsequent transaction signing phase may be considered malicious.
```

Comment:
```suggestion
Failure to confirm during input registration can generally be tolerated as it does not harm other users. However, failure to confirm in connection confirmation phase cannot go without punishment, otherwise an attacker could make sure only small coinjoins to take place. And once the connection confirmation phase times out the input set is finalized, and inputs for which a signature is not the subsequent transaction signing phase may be considered malicious.
```